schema/scan.go

```diff
@@ -449,6 +449,11 @@ func PtrScanner(fn ScannerFunc) ScannerFunc {
 		if dest.IsNil() {
 			dest.Set(reflect.New(dest.Type().Elem()))
 		}
+
+		if dest.Kind() == reflect.Map {
+			return fn(dest, src)
+		}
```

Comment:
This looks confusing and wrong. Do we need it? Is there any other way to achieve the same?

Reply:
Well, this covers a pretty odd case, but which I came across in my expirience - usage of `*map[string]string` (there is a test on this in pg_test.go).
Without this it causes a `panic: reflect: call of reflect.Value.Elem on map Value`

Reply:
Could you please make sure we have a test for that case so we don't break it accidentally?

Reply:
Yeah, there is a test on this [here](https://github.com/uptrace/bun/pull/567/files#diff-1475aeec71b3cf89192f35901eb4162d64cc2875904261d1cbcfe60363ed613aR544-R550)

Reply:
:+1: